icu4c/source/common/edits.cpp

```diff
@@ -482,7 +482,6 @@ UBool Edits::Iterator::next(UBool onlyChanges, UErrorCode &errorCode) {
     int32_t u = array[index++];
     if (u <= MAX_UNCHANGED) {
         // Combine adjacent unchanged ranges.
-        changed = FALSE;
```

Comment:
This looks bad. `changed` is a member variable, and when it's not set right, then the observable iterator state is wrong.